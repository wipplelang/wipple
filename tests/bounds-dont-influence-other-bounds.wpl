T : A => trait

instance (T Number)
instance (T Text)

f :: A B where (T A) (T B) => (A ; B)
f : ...

(f) :: (Number ; _) -- `B` should not be inferred to be `Number`
