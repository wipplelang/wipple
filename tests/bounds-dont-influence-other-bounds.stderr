error: 
  ┌─ test:9:2
  │
9 │ (f) :: (Number , _) -- `B` should not be inferred to be `Number`
  │  ^
  │  │
  │  could not determine the type of this expression
  │  this has type `Number , _`
  │
  = annotate the type with `::`: `:: {%type%}`
  = for more information, see https://wipple.dev/playground/?lesson=errors/unknown-type

